.github/workflows/pr-slack-notification.yaml

```diff
@@ -0,0 +1,304 @@
+            pr_title=$(_jq '.title')
+            pr_user=$(_jq '.user.login')
+            pr_url=$(_jq '.html_url')
+            pr_closed_at=$(_jq '.closed_at')
+            pr_created_at=$(_jq '.created_at')
+
+            if [[ "$pr_closed_at" > "$last_run_timestamp" ]]; then
+              closed_pr_list="${closed_pr_list}\n*Closed Pull Request* #${pr_number}: ${pr_title}\n*Closed by*: ${pr_user}\n*URL*: ${pr_url}\n*Closed At*: ${pr_closed_at}\n"
+            fi
+          done
+          
+          echo "Latest Closed PR List ------$closed_pr_list"
+
+          if [ -n "$closed_pr_list" ]; then
+            echo -e "$closed_pr_list" > closed_pr_list.txt
+            echo "::set-output name=notify_closed::true"
+          else
+            echo "::set-output name=notify_closed::false"
+          fi      
+
+      - name: Send Slack notification for PRs
+        if: success() && (steps.fetch_all_pull_requests_using_condition.outputs.notify == 'true' || steps.fetch_closed_prs.outputs.notify_closed == 'true')
+        run: |
+          # Initialize PR lists
+          pr_list=""
+          closed_pr_list=""
+
+          # Check if the open PRs file exists and read its content
+          if [ -f pr_list.txt ]; then
+            pr_list=$(cat pr_list.txt)
+          fi
+
+          # Check if the closed PRs file exists and read its content
+          if [ -f closed_pr_list.txt ]; then
+            closed_pr_list=$(cat closed_pr_list.txt)
+          fi
+
+          # Initialize payload blocks
+          payload_blocks=()
+
+          # Add open PRs section if not empty
+          if [ -n "$pr_list" ]; then
+            payload_blocks+=("{
+              \"type\": \"header\",
+              \"text\": {
+                \"type\": \"plain_text\",
+                \"text\": \"List of Open/New/Updated Pull Requests using Cron Job\"
+              }
+            }")
+            payload_blocks+=("{
+              \"type\": \"section\",
+              \"text\": {
+                \"type\": \"mrkdwn\",
+                \"text\": \"$pr_list\"
+              }
+            }")
+          fi
+
+          # Add closed PRs section if not empty
+          if [ -n "$closed_pr_list" ]; then
+            payload_blocks+=("{
+              \"type\": \"header\",
+              \"text\": {
+                \"type\": \"plain_text\",
+                \"text\": \"List of Pull Requests Closed Since the Last Cron Job\"
+              }
+            }")
+            payload_blocks+=("{
+              \"type\": \"section\",
+              \"text\": {
+                \"type\": \"mrkdwn\",
+                \"text\": \"$closed_pr_list\"
+              }
+            }")
+          fi
+
+          # Construct the payload
+          payload=$(cat <<-EOF
+          {
+            "blocks": [
+              $(IFS=,; echo "${payload_blocks[*]}")
+            ]
+          }
+          EOF
+          )
+
+          # Debugging: Print payload to verify its format
```

Comment:
Remove print statements used for debugging.

Reply:
Yes I removed it.